### PR TITLE
fix: :bug: Match the UI and API path for NotificationSource -> source

### PR DESCRIPTION
**Pull request by chadell**
Breaking Change: API endpoint for notificationsource should be changed to source

Related to #313

### CODE REVIEW

File: nautobot_circuit_maintenance/api/urls.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 router.register("maintenance", views.MaintenanceTaskView)
 router.register("note", views.MaintenanceNoteTaskView)
 router.register("circuitimpact", views.MaintenanceCircuitImpactTaskView)
-router.register("notificationsource", views.NotificationSourceTaskView)
+router.register("source", views.NotificationSourceTaskView)
 router.register("parsednotification", views.ParsedNotificationTaskView)
 router.register("rawnotification", views.RawNotificationTaskView)
```
Comment (glennmatthews) [inline nautobot_circuit_maintenance/api/urls.py:13]: If we're making breaking changes anyway, *all* of these are wrong - should be `maintenances`, `notes`, `circuit-impacts`, `sources`, `parsed-notifications`, `raw-notifications`, etc.

Comment (chadell) [reply nautobot_circuit_maintenance/api/urls.py:13]: nice to know, where is this convention defined?

File: nautobot_circuit_maintenance/urls.py [hunk 2]
```diff
@@ -10,80 +10,80 @@
...
+    path("notification-sources/<uuid:pk>/validate/", views.NotificationSourceValidate.as_view(), name="notificationsource_validate"),
     path(
-        "source/<uuid:pk>/changelog/",
+        "notification-sources/<uuid:pk>/changelog/",
         ObjectChangeLogView.as_view(),
         name="notificationsource_changelog",
         kwargs={"model": NotificationSource},
     ),
-    path("docs/", RedirectView.as_view(url=static("nautobot_circuit_maintenance/docs/index.html")), name="docs"),
+    path("docs/", RedirectView.as_view(url=static("nautobot_circuit_maintenances/docs/index.html")), name="docs"),
```
Comment (glennmatthews) [inline nautobot_circuit_maintenance/urls.py:88]: revert?

File: nautobot_circuit_maintenance/urls.py [hunk 3]
```diff
@@ -10,80 +10,80 @@
 
 urlpatterns = [
     # Overview
-    path("maintenance/overview/", views.CircuitMaintenanceOverview.as_view(), name="circuitmaintenance_overview"),
+    path("maintenances/overview/", views.CircuitMaintenanceOverview.as_view(), name="circuitmaintenance_overview"),
```
Comment (glennmatthews) [inline nautobot_circuit_maintenance/urls.py:13]: `circuit-maintenances` to match API?

**Review by glennmatthews — APPROVED** [review]
LGTM - thanks!

**Comment by gsnider2195** [comment]
Since this is a breaking change should we bump the pyproject.toml version so that when the next release is prepped for this project we don't forget to bump the major version?

File: changes/320.fixed [hunk 4]
```diff
@@ -0,0 +1 @@
+Homogenized the path for NotificationSources to `source` in UI and API
```
Comment (gsnider2195) [inline changes/320.fixed:1]:
```suggestion
Homogenized the path for NotificationSources to `source` in UI and API.
```

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**